app/gui/view/documentation/src/lib.rs

```diff
@@ -173,17 +181,33 @@ impl Model {
     }
 
     /// Display the documentation and scroll to the qualified name if needed.
-    fn display_doc(&self, docs: EntryDocumentation) {
-        let anchor = docs.function_name().map(html::anchor_name);
-        let html = html::render(docs);
+    fn display_doc(&self, docs: EntryDocumentation, display_doc: &frp::Source<EntryDocumentation>) {
+        let linked_pages = docs.linked_doc_pages();
+        let html = html::render(&docs);
         self.inner_dom.dom().set_inner_html(&html);
-        if let Some(anchor) = anchor {
+        self.set_link_handlers(linked_pages, display_doc);
+        // Scroll to the top of the page.
+        self.inner_dom.dom().set_scroll_top(0);
+    }
+
+    /// Setup event handlers for links on the documentation page.
+    fn set_link_handlers(
+        &self,
+        linked_pages: Vec<LinkedDocPage>,
+        display_doc: &frp::Source<EntryDocumentation>,
+    ) {
+        let mut handlers = self.event_handlers.borrow_mut();
+        handlers.clear();
+        for page in linked_pages {
+            let content = page.page.clone_ref();
+            let anchor = html::anchor_name(&page.name);
             if let Some(element) = web::document.get_element_by_id(&anchor) {
-                let offset = element.dyn_ref::<HtmlElement>().map(|e| e.offset_top()).unwrap_or(0);
-                self.inner_dom.dom().set_scroll_top(offset);
+                let closure: web::JsEventHandler = web::Closure::new(f_!([display_doc, content] {
+                    display_doc.emit(content.clone_ref());
+                }));
+                let handler = web::add_event_listener(&element, "click", closure);
+                handlers.push(handler);
             }
```

Comment:
What about `handlers.replace(linked_pages.filter_map(...).collect())`?

Reply:
I doubt it is more readable, but I changed it.